src/reducers/versions.tsx

```diff
@@ -424,29 +429,64 @@ export const createInternalVersion = (
   };
 };
 
-export const getVersionFiles = (versions: VersionsState, versionId: number) => {
-  return versions.versionFiles[versionId];
+type GetVersionEntriesKeyParams = {
+  versionId: number;
+  comparedToVersionId: number | null | undefined;
+};
+
+export const getVersionEntriesKey = ({
+  versionId,
+  comparedToVersionId,
+}: GetVersionEntriesKeyParams) => {
+  if (comparedToVersionId === undefined) {
+    return `versionId=${versionId}`;
+  }
+  return `versionId=${versionId};comparedToVersionId=${comparedToVersionId}`;
+};
+
+export const createInternalVersionEntries = (
+  version: ExternalVersionWithContent | ExternalVersionWithDiff,
+) => {
+  return Object.keys(version.file.entries).map((nodeName) => {
+    return createInternalVersionEntry(version.file.entries[nodeName]);
+  });
 };
 
-export const getVersionInfo = (
+export const getVersionEntries = (
   versions: VersionsState,
-  versionId: number,
-  { comparedToVersionId }: { comparedToVersionId?: number } = {},
+  {
+    versionId,
+    comparedToVersionId,
+  }: {
+    versionId: number;
+    comparedToVersionId:
+      | undefined // To get versionEntries with any comparedToVersionId
+      | null // To get versionEntries with comparedToVersionId is null
+      | number;
+  },
 ) => {
-  const version = versions.versionInfo[versionId];
+  const { versionEntries } = versions;
 
-  if (version === undefined) {
-    return undefined;
-  }
-  if (
-    version &&
-    comparedToVersionId !== undefined &&
-    version.comparedToVersionId !== comparedToVersionId
-  ) {
-    return undefined;
+  const key = getVersionEntriesKey({ versionId, comparedToVersionId });
+
+  if (comparedToVersionId === undefined) {
```

Comment:
You shouldn't need to special case `undefined` values because they are already part of the key. Try commenting out this if statement then see if you can get all tests to pass. It will involve some changes, such as passing `comparedToVersionId` into `getVersionFile()` (maybe?) but I think we need those changes. 

I realize that this if statement was trying to implicitly support version entries that had no relation to a version-comparison but we can do that *explicitly* now by passing `comparedToVersionId: undefined` (or `null`). To put it another way, the fact that you needed this if statement makes me think the reducer functions aren't designed correctly yet.

Reply:
I updated the patch to pass `comparedToVersionId` to some functions, like `getVersionFile`, `expandTree` and `setVisibleSelectedPath`. It is a little annoying to have to pass this "irrelevant" parameter, since these functions did not use `versionEntry.status` at all which is the only property depending on `comparedToVersionId`. But I guess this way make the code more robust than the previous one.